aptos-move/aptos-transactional-test-harness/src/aptos_test_harness.rs

```diff
@@ -0,0 +1,943 @@
+        if let Some(private_key) = self.private_key_mapping.get(s.as_str()) {
+            return private_key.clone();
+        }
+        panic!("Failed to resolve private key '{}'", s)
+    }
+
+    /// Resolve a raw private key into a numeric one.
+    fn resolve_private_key(&self, private_key: &RawPrivateKey) -> Ed25519PrivateKey {
+        match private_key {
+            RawPrivateKey::Anonymous(private_key) => private_key.clone(),
+            RawPrivateKey::Named(name) => self.resolve_named_private_key(name),
+        }
+    }
+
+    /// Resolve addresses and private keys for secondary signers.
+    fn resolve_secondary_signers(
+        &mut self,
+        secondary_signers: &[SignerAndKeyPair],
+    ) -> (Vec<AccountAddress>, Vec<Ed25519PrivateKey>) {
+        let mut addresses = vec![];
+        let mut private_keys = vec![];
+
+        for SignerAndKeyPair {
+            address,
+            private_key,
+        } in secondary_signers
+        {
+            addresses.push(self.compiled_state().resolve_address(address));
+
+            let resolved_private_key = match (private_key, address) {
+                (Some(private_key), _) => self.resolve_private_key(private_key),
+                (None, ParsedAddress::Named(named_addr)) => {
+                    match self.private_key_mapping.get(named_addr) {
+                        Some(private_key) => private_key.clone(),
+                        None => panic!(
+                            "Failed to resolve private key for secondary signer {}.",
+                            named_addr
+                        ),
+                    }
+                }
+                (None, ParsedAddress::Numerical(addr)) => {
+                    panic!("No private key provided for secondary signer {}.", addr)
+                }
+            };
+
+            private_keys.push(resolved_private_key);
+        }
+
+        (addresses, private_keys)
+    }
+
+    /// Obtain a Rust representation of the account resource from storage, which is used to derive
+    /// a few default transaction parameters.
+    fn fetch_account_resource(&self, signer_addr: &AccountAddress) -> Result<AccountResource> {
+        let account_access_path = AccessPath::resource_access_path(ResourceKey::new(
+            *signer_addr,
+            AccountResource::struct_tag(),
+        ));
+        let account_blob = self
+            .storage
+            .get_state_value(&StateKey::AccessPath(account_access_path))
+            .unwrap()
+            .ok_or_else(|| {
+                format_err!(
+                "Failed to fetch account resource under address {}. Has the account been created?",
+                signer_addr
+            )
+            })?;
+        Ok(bcs::from_bytes(&account_blob).unwrap())
+    }
+
+    /// Obtain the TestCoin amount under address `signer_addr`
+    fn fetch_account_balance(&self, signer_addr: &AccountAddress) -> Result<u64> {
```

Comment:
We should definitely be using the types instead of this: https://github.com/aptos-labs/aptos-core/tree/main/types/src/account_config/resources

same for the resource above

Reply:
updated